test/framework/src/main/java/org/opensearch/test/ParameterizedOpenSearchIntegTestCase.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test;
+
+import org.junit.After;
+import org.junit.Before;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Base class for running the tests with parameterization of the dynamic settings
+ * For any class that wants to use parameterization, use @ParametersFactory to generate
+ * different params only for dynamic settings. Refer SearchCancellationIT for an example.
+ * Note: this doesn't work for the parameterization of feature flag/static settings.
+ */
+public abstract class ParameterizedOpenSearchIntegTestCase extends OpenSearchIntegTestCase {
+
+    private final Settings dynamicSettings;
+
+    public ParameterizedOpenSearchIntegTestCase(Settings dynamicSettings) {
+        this.dynamicSettings = dynamicSettings;
+    }
+
+    @Before
+    public void beforeTests() {
+        client().admin().cluster().prepareUpdateSettings().setPersistentSettings(dynamicSettings).get();
+    }
+
+    @After
+    public void afterTests() {
+        client().admin().cluster().prepareUpdateSettings().setPersistentSettings(Settings.builder().putNull("*")).get();
```

Comment:
Is this safe or will it possibly stomp other settings? How about something like:

```
final Settings.Builder settingsToUnset = Settings.builder();
dynamicSettings.names().forEach(settingsToUnset::putNull);
client().admin().cluster().prepareUpdateSettings().setPersistentSettings(settingsToUnset).get();
```

?

Reply:
@andrross there are multiple places we are using the settings unset like the one used here, ref: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/search/SearchCancellationIT.java#L112

Reply:
on second thoughts, there could be some setting that could be shared across multiple tests in a class so you are right we should only reset the settings which we are setting in this class, will make this change.

Reply:
dynamicSettings.names() isnt right so using this instead
```
dynamicSettings.keySet().forEach(settingsToUnset::putNull)
```